### PR TITLE
[PWGHF] Developments for charm resonances workflow

### PR DESCRIPTION
**Pull request by Luca610**
1. dataCreatorCharmResoReduced.cxx:  
- added decay channel of Ds1 with at least one muon in the final state
- changed debug histogram ranges

2. candidateCreatorCharmResoReduced.cxx: 
- Removed indices from output table to avoid unbound indices in slim derived data
- Moved Mixed event here
- Added debug histograms to MC reconstruction analysis

3. taskCharmResoReduced:
- Added output tree for offline analysis
- Added MC process functions

### CODE REVIEW

File: PWGHF/D2H/TableProducer/candidateCreatorCharmResoReduced.cxx [hunk 1]
```diff
@@ -332,27 +348,121 @@ struct HfCandidateCreatorCharmResoReduced {
           rowCandidateResoTrack(pVecD[0], pVecD[1], pVecD[2],
                                 candV0Tr.px(), candV0Tr.py(), candV0Tr.pz(),
                                 invMassReso,
-                                invMassD);
+                                invMassD - invMassD0);
```
Comment (fgrosa) [inline PWGHF/D2H/TableProducer/candidateCreatorCharmResoReduced.cxx:351]: it seems to me from line 182 and 184 that `invMassD` already is the delta mass (so here we are subtracting the D0 mass twice). Personally I would keep the `invMassD` without the D0 mass subtraction and I would define another variable `deltaInvMass` to be used when needed for clarity.

File: PWGHF/D2H/TableProducer/candidateCreatorCharmResoReduced.cxx [hunk 2]
```diff
@@ -332,27 +348,121 @@ struct HfCandidateCreatorCharmResoReduced {
...
         } else {
-          rowCandidateReso(collision.globalIndex(),
-                           candD.globalIndex(),
-                           candV0Tr.globalIndex(),
-                           pVecD[0], pVecD[1], pVecD[2],
+          rowCandidateReso(pVecD[0], pVecD[1], pVecD[2],
                            pVecV0Tr[0], pVecV0Tr[1], pVecV0Tr[2],
                            invMassReso,
                            invMassD,
```
Comment (fgrosa) [inline PWGHF/D2H/TableProducer/candidateCreatorCharmResoReduced.cxx:356]: now this is the delta mass, but I would put the mass instead (see comment above)

File: PWGHF/D2H/TableProducer/candidateCreatorCharmResoReduced.cxx [hunk 3]
```diff
@@ -332,27 +348,121 @@ struct HfCandidateCreatorCharmResoReduced {
...
+            registry.fill(HIST("hMassLambdaDminus"), invMassReso, ptReso);
+            break;
+          default:
+            break;
+        }
+        // Fill output table
+        rowCandidateReso(pVecD[0], pVecD[1], pVecD[2],
+                         pVecV0Tr[0], pVecV0Tr[1], pVecV0Tr[2],
+                         invMassReso,
+                         invMassD,
```
Comment (fgrosa) [inline PWGHF/D2H/TableProducer/candidateCreatorCharmResoReduced.cxx:449]: same here

File: PWGHF/D2H/TableProducer/candidateCreatorCharmResoReduced.cxx [hunk 4]
```diff
@@ -515,22 +674,31 @@ struct HfCandidateCreatorCharmResoReducedExpressions {
         filledMcInfo = true;
         if (TESTBIT(rowDV0McRec.flagMcMatchRec(), DecayTypeMc::Ds1ToDStarK0ToD0PiK0s) || TESTBIT(rowDV0McRec.flagMcMatchRec(), DecayTypeMc::Ds2StarToDplusK0sToPiKaPiPiPi)) {
           registry.fill(HIST("hMassMcMatched"), candReso.invMass(), candReso.pt());
+          registry.fill(HIST("hMassMcMatchedVsBach0Mass"), candReso.invMass(), candReso.invMassProng0() - candReso.invMassD0());
```
Comment (fgrosa) [inline PWGHF/D2H/TableProducer/candidateCreatorCharmResoReduced.cxx:677]: currently `invMassProng0` already returns the delta mass (see above)

File: PWGHF/D2H/TableProducer/candidateCreatorCharmResoReduced.cxx [hunk 5]
```diff
@@ -515,22 +674,31 @@ struct HfCandidateCreatorCharmResoReducedExpressions {
...
+          registry.fill(HIST("hMassMcMatchedVsBach1Mass"), candReso.invMass(), candReso.invMassProng1());
+          registry.fill(HIST("hMassMcMatchedVsD0Mass"), candReso.invMass(), candReso.invMassD0());
+
         } else if (TESTBIT(rowDV0McRec.flagMcMatchRec(), DecayTypeMc::Ds1ToDStarK0ToDPlusGammaK0s) || TESTBIT(rowDV0McRec.flagMcMatchRec(), DecayTypeMc::Ds1ToDStarK0ToDPlusPi0K0s)) {
           registry.fill(HIST("hMassMcMatchedIncomplete"), candReso.invMass(), candReso.pt());
         } else {
           registry.fill(HIST("hMassMcUnmatched"), candReso.invMass(), candReso.pt());
+          registry.fill(HIST("hMassMcUnmatchedVsBach0Mass"), candReso.invMass(), candReso.invMassProng0() - candReso.invMassD0());
```
Comment (fgrosa) [inline PWGHF/D2H/TableProducer/candidateCreatorCharmResoReduced.cxx:685]: currently `invMassProng0` already returns the delta mass (see above)

**Review by fgrosa — CHANGES_REQUESTED** [review]
Hi @Luca610 thanks!
The PR is huge and I hope that I didn't miss anything, but I think I spot a bug, please take a look

**Comment by Luca610** [comment]
Hi @fgrosa, thanks for the review and sorry for the large PR, I forgot to split different developments. The bug you mention looks correct to me, the difference in line 184 is in a function separated from the main one

**Review by fgrosa — APPROVED** [review]

**Pull request merged**